Improve routing documentation (#5450)
containous/traefik: docs/content/routing/entrypoints.md

            graceTimeOut: 42
          respondingTimeouts:
            readTimeout: 42
            writeTimeout: 42
            idleTimeout: 42
        proxyProtocol:
          insecure: true
          trustedIPs:
          - "127.0.0.1"
          - "192.168.0.1"
        forwardedHeaders:
          insecure: true
          trustedIPs:
          - "127.0.0.1"
          - "192.168.0.1"
    ```
    
    ```bash tab="CLI"
    ## Static configuration
    --entryPoints.name.address=:8888
    --entryPoints.name.transport.lifeCycle.requestAcceptGraceTimeout=42
    --entryPoints.name.transport.lifeCycle.graceTimeOut=42
    --entryPoints.name.transport.respondingTimeouts.readTimeout=42
    --entryPoints.name.transport.respondingTimeouts.writeTimeout=42
    --entryPoints.name.transport.respondingTimeouts.idleTimeout=42
    --entryPoints.name.proxyProtocol.insecure=true
    --entryPoints.name.proxyProtocol.trustedIPs="127.0.0.1,192.168.0.1"
    --entryPoints.name.forwardedHeaders.insecure=true
    --entryPoints.name.forwardedHeaders.trustedIPs="127.0.0.1,192.168.0.1"
    ```

### Forwarded Header

You can configure Traefik to trust the forwarded headers information (`X-Forwarded-*`).

> ??? info "`forwardedHeaders.trustedIPs`"

Reviewer: ```suggestion
??? info "`forwardedHeaders.trustedIPs`: Trusting Forwarded Headers from specific IPs."
```

Author: I prefer to be homogeneous with other element.
The name of the option can be long.

>     Trusting Forwarded Headers from specific IPs.

Reviewer: ```suggestion
```

Author: I prefer to be homogeneous with other element.
The name of the option can be long.

    ```toml tab="File (TOML)"
    ## Static configuration
    [entryPoints]
      [entryPoints.web]
        address = ":80"
    
        [entryPoints.web.forwardedHeaders]
          trustedIPs = ["127.0.0.1/32", "192.168.1.7"]
    ```
    
    ```yaml tab="File (YAML)"
    ## Static configuration
    entryPoints:
      web:
        address: ":80"
        forwardedHeaders:
          trustedIPs:
          - "127.0.0.1/32"
          - "192.168.1.7"
    ```
    
    ```bash tab="CLI"
    ## Static configuration
    --entryPoints.web.address=:80
    --entryPoints.web.forwardedHeaders.trustedIPs=127.0.0.1/32,192.168.1.7
    ```

> ??? info "`forwardedHeaders.insecure`"

Reviewer: ```suggestion
??? info "`forwardedHeaders.insecure`: Insecure Mode (Always Trusting Forwarded Headers)."
```

Author: I prefer to be homogeneous with other element.
The name of the option can be long.

>     Insecure Mode (Always Trusting Forwarded Headers).

Reviewer: ```suggestion
```

Author: I prefer to be homogeneous with other element.
The name of the option can be long.

    --entryPoints.name.address=:8888
    --entryPoints.name.transport.lifeCycle.requestAcceptGraceTimeout=42
    ```

??? info "`lifeCycle.graceTimeOut`"
    
    _Optional, Default=10s_
    
    Duration to give active requests a chance to finish before Traefik stops.
    
    Can be provided in a format supported by [time.ParseDuration](https://golang.org/pkg/time/#ParseDuration) or as raw values (digits).
    
    If no units are provided, the value is parsed assuming seconds.
    
    !!! warning "In this time frame no new requests are accepted."
    
    ```toml tab="File (TOML)"
    ## Static configuration
    [entryPoints]
      [entryPoints.name]
        address = ":8888"
        [entryPoints.name.transport]
          [entryPoints.name.transport.lifeCycle]
            graceTimeOut = 42
    ```
    
    ```yaml tab="File (YAML)"
    ## Static configuration
    entryPoints:
      name:
        address: ":8888"
        transport:
          lifeCycle:
            graceTimeOut: 42
    ```
    
    ```bash tab="CLI"
    ## Static configuration
    --entryPoints.name.address=:8888
    --entryPoints.name.transport.lifeCycle.graceTimeOut=42
    ```

### ProxyProtocol

> Traefik supports [ProxyProtocol](https://www.haproxy.org/download/2.0/doc/proxy-protocol.txt) version 1 and 2.

Reviewer: I'm not sure if ProxyProtocol works with TCP routers/services (e.g. "can I enable ProxyProtocol for a MySQL server behind traefik?").

- If it is not, we have to add a "warning" admonition here.
- If it is supported, then this comment can be dismissed.

Author: out of the scope of my PR for now.

> If proxyprotocol header parsing is enabled for the entry point, this entry point can accept connections with or without proxyprotocol headers.

Reviewer: ```suggestion
If ProxyProtocol header parsing is enabled for the entrypoint, this entrypoint can accept connections with or without ProxyProtocol headers.
```

> If the proxyprotocol header is passed, then the version is determined automatically.

Reviewer: ```suggestion
If the ProxyProtocol header is passed, then the version is determined automatically.
```

> ??? info "`proxyProtocol.trustedIPs`" 

Reviewer: ```suggestion
??? info "`proxyProtocol.trustedIPs`: Enabling Proxy Protocol with Trusted IPs." 
```

Author: I prefer to be homogeneous with other element.
The name of the option can be long.

>     Enabling Proxy Protocol with Trusted IPs.

Reviewer: ```suggestion
```

Author: I prefer to be homogeneous with other element.
The name of the option can be long.

    ```toml tab="File (TOML)"
    ## Static configuration
    [entryPoints]
      [entryPoints.web]
        address = ":80"
    
        [entryPoints.web.proxyProtocol]
          trustedIPs = ["127.0.0.1/32", "192.168.1.7"]
    ```
    
    ```yaml tab="File (YAML)"
    ## Static configuration
    entryPoints:
      web:
        address: ":80"
        proxyProtocol:
          trustedIPs:
          - "127.0.0.1/32"
          - "192.168.1.7"
    ```
    
    ```bash tab="CLI"
    --entryPoints.web.address=:80
    --entryPoints.web.proxyProtocol.trustedIPs=127.0.0.1/32,192.168.1.7
    ```

    IPs in `trustedIPs` only will lead to remote client address replacement: Declare load-balancer IPs or CIDR range here.

> ??? info "`proxyProtocol.insecure`"

Reviewer: ```suggestion
??? info "`proxyProtocol.insecure`: Insecure Mode (Testing Environment Only)."
```

Author: I prefer to be homogeneous with other element.
The name of the option can be long.

>     Insecure Mode (Testing Environment Only).

Reviewer: ```suggestion
```

Author: I prefer to be homogeneous with other element.
The name of the option can be long.
